File: src/main/java/com/hoi4utils/clausewitz/data/technology/Technology.java
package com.hoi4utils.clausewitz.data.technology;

import com.hoi4utils.clausewitz.code.ClausewitzDate;
import com.hoi4utils.clausewitz.localization.*;
import org.jetbrains.annotations.NotNull;
import scala.jdk.javaapi.CollectionConverters;

import java.util.Collection;
import java.util.List;
import java.util.Map;

public class Technology implements Localizable {
	public String id;
	public ClausewitzDate technologyDate;
	public double cost;    // research_cost
	public int year;    // start_year
	public List<Equipment> enabledEquipments;
	public List<EquipmentModule> enabledEquipmentModules;
	public List<TechCategory> categories;
	public List<TechPath> paths;
	// ai_will_do

	@Override
	public @NotNull scala.collection.mutable.Map<Property, String> getLocalizableProperties() {
		return CollectionConverters.asScala(Map.of(
			Property.NAME, id
		)); 
	}

	// todo this can be more inclusive
	@Override
	public @NotNull scala.collection.Iterable<? extends Localizable> getLocalizableGroup() {
		return CollectionConverters.asScala(List.of(this));
	}
}

File: src/main/scala/com/hoi4utils/hoi4/technology/Technology.scala
package com.hoi4utils.hoi4.technology

import com.hoi4utils.ClausewitzDate
import com.hoi4utils.localization.{Localizable, Property}
import org.jetbrains.annotations.NotNull

import scala.collection.mutable
import scala.collection.mutable.Map
import scala.jdk.javaapi.CollectionConverters


class Technology extends Localizable {
  var id: String = "" 
  var technologyDate: Option[ClausewitzDate] = None
  var cost = .0 // research_cost

  var year = 0 // start_year

  var enabledEquipments: List[Equipment] = Nil
  var enabledEquipmentModules: List[EquipmentModule] = Nil
  var categories:List[TechCategory] = Nil
  var paths: List[TechPath] = Nil

  // ai_will_do
  @NotNull override def getLocalizableProperties: mutable.Map[Property, String] = mutable.Map(Property.NAME -> id)

  // todo this can be more inclusive
  @NotNull override def getLocalizableGroup: Iterable[? <: Localizable] = List(this)
}
